src/commands/exec.c

```diff
@@ -20,20 +20,132 @@
+ * Only one is required ( container id )
+ * exec can be done using:
+ * - runtime exec [options] container_id cmd args
+ * - runtime exec [options] -p process.json container_id
+ * where process.json is have a
+ * json node like \c oci_cfg_process
+ **/
+#define MIN_EXEC_ARGS 1
+
+extern struct start_data start_data;
+
+gchar    *cwd          = NULL;
+gchar   **env          = NULL;
+gchar    *process_json = NULL;
+
+/* Not supported flags */
+/* clear containers does not support apparmor*/
+gchar    *apparmor       = NULL;
+gchar    *cap            = NULL;
+/* clear containers does not support SELINUX*/
+gchar    *process_label  = NULL;
+gboolean  no_new_privs   = false;
+gboolean *no_subreaper   = NULL;
+
+
+/* ignore -pedantic to cast handle_option_console
+ * and  handle_option_user, a function pointer, to a
+ * void* */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+static GOptionEntry options_exec[] =
+{
+	{
+		"apparmor", 0, G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_STRING, &apparmor,
+		/* clear containers does not support apparmor */
+		"not implemented",
+		NULL
+	},
+	{
+		"cap", 'c', G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_STRING, &cap,
+		"not implemented",
+		NULL
+	},
+	{
+		"console", 0, G_OPTION_FLAG_OPTIONAL_ARG,
+		G_OPTION_ARG_CALLBACK, handle_option_console,
+		"set pty console that will by the exec workload",
+		NULL
+	},
+	{
+		"cwd", 0, G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_STRING, &cwd,
+		"current working directory to run the exec workload",
+		NULL
+	},
+	{
+		"detach", 'd' , G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_NONE, &start_data.detach,
+		"exec process in detach mode",
+		NULL
+	},
+	{
+		"env", 'e', G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_STRING_ARRAY, &env,
+		"in the container",
+		NULL
+	},
+	{
+		"no-new-privs", 0, G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_NONE, &no_new_privs,
+		"not implemented",
+		NULL
+	},
+	{
+		"no-subreaper", 0, G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_NONE, &no_subreaper,
+		"not implemented",
+		NULL
+	},
+	{
+		"pid-file", 0, G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_STRING, &start_data.pid_file,
+		"the file to write the process ID of the new "
+		"process executed in the container",
+		NULL
+	},
+	{
+		"process", 'p' , G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_STRING, &process_json,
+		"specify path to process.json",
+		NULL
+	},
+	{
+		"tty", '0' , G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_NONE, &start_data.allocate_tty,
+		"allocate a pseudo-TTY for the new exec process",
+		NULL
+	},
+	{
+		"user", 'u' , G_OPTION_FLAG_NONE,
+		G_OPTION_ARG_CALLBACK, handle_option_user,
+		"<uid>[:<gid>], UID for the process to run in format",
+		NULL
+	},
+	{NULL}
+};
+
 static gboolean
 handler_exec (const struct subcommand *sub,
 		struct cc_oci_config *config,
 		int argc, char *argv[])
 {
 	struct oci_state  *state = NULL;
 	gchar             *config_file = NULL;
-	gboolean           ret;
+	gboolean           ret = false;
+	struct oci_cfg_process *process = &config->oci.process;
```

Comment:
Dereferencing `config` before checking it. I'd be tempted to replace the asserts below with `if (...)` tests instead, then assign `process.